gdnative-sys bindgen cannot find 'TargetConditionals.h' on MacOS

gdnative-sys bindgen fails on MacOS with the error `godot-rust/sys/godot_headers/gdnative/gdnative.h:42:10: fatal error: 'TargetConditionals.h' file not found`.

--- a/examples/hello_world/hello_world_library.gdnlib
+++ b/examples/hello_world/hello_world_library.gdnlib
@@ -1,10 +1,12 @@
 [entry]
 
 X11.64="../../target/debug/libhello_world.so"
+OSX.64="../../target/debug/libhello_world.dylib"
 
 [dependencies]
 
 X11.64=[  ]
+OSX.64=[  ]
 
 [general]
 

--- a/examples/manually_registered/manually_registered_library.gdnlib
+++ b/examples/manually_registered/manually_registered_library.gdnlib
@@ -1,10 +1,12 @@
 [entry]
 
 X11.64="../../target/debug/libmanually_registered.so"
+OSX.64="../../target/debug/libmanually_registered.dylib"
 
 [dependencies]
 
 X11.64=[  ]
+OSX.64=[  ]
 
 [general]
 

--- a/examples/spinning_cube/spinning_cube_library.gdnlib
+++ b/examples/spinning_cube/spinning_cube_library.gdnlib
@@ -1,10 +1,12 @@
 [entry]
 
 X11.64="../../target/debug/libspinning_cube.so"
+OSX.64="../../target/debug/libspinning_cube.dylib"
 
 [dependencies]
 
 X11.64=[  ]
+OSX.64=[  ]
 
 [general]
 

--- a/sys/build.rs
+++ b/sys/build.rs
@@ -4,20 +4,59 @@ extern crate bindgen;
 use std::env;
 use std::path::PathBuf;
 
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+fn osx_include_path() -> Result<String, std::io::Error> {
+    use std::process::Command;
+
+    let output = Command::new("xcode-select").arg("-p").output()?.stdout;
+    let prefix_str = std::str::from_utf8(&output).expect("invalid output from `xcode-select`");
+    let prefix = prefix_str.trim_right();
+
+    let platform = if cfg!(target_os = "macos") {
+        "MacOSX"
+    } else if cfg!(target_os = "ios") {
+        "iPhoneOS"
+    } else {
+        unreachable!();
+    };
+
+    let infix = if prefix == "/Library/Developer/CommandLineTools" {
+        format!("SDKs/{}.sdk", platform)
+    } else {
+        format!("Platforms/{}.platform/Developer/SDKs/{}.sdk", platform, platform)
+    };
+
+    let suffix = "usr/include";
+    let directory = format!("{}/{}/{}", prefix, infix, suffix);
+
+    Ok(directory)
+}
+
 fn main() {
     let dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let bindings = bindgen::Builder::default()
-        .header("godot_headers/gdnative_api_struct.gen.h")
-        .whitelisted_type("godot.*")
-        .whitelisted_function("godot.*")
-        .whitelisted_var("godot.*")
-        .whitelisted_type("GDNATIVE.*")
-        .derive_default(true)
-        .ignore_functions()
-        .ctypes_prefix("libc")
-        .clang_arg(format!("-I{}/godot_headers", dir))
-        .generate()
-        .expect("Unable to generate bindings");
+    let bindings = {
+        let mut builder = bindgen::Builder::default()
+            .header("godot_headers/gdnative_api_struct.gen.h")
+            .whitelisted_type("godot.*")
+            .whitelisted_function("godot.*")
+            .whitelisted_var("godot.*")
+            .whitelisted_type("GDNATIVE.*")
+            .derive_default(true)
+            .ignore_functions()
+            .ctypes_prefix("libc")
+            .clang_arg(format!("-I{}/godot_headers", dir));
+
+        #[cfg(any(target_os = "macos", target_os = "ios"))]
+        match osx_include_path() {
+            Ok(osx_include_path) => {
+                builder = builder.clang_arg("-I").clang_arg(osx_include_path);
+            },
+            _ => {},
+        }
+
+        builder.generate()
+            .expect("Unable to generate bindings")
+    };
 
     let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
     bindings

